tests/backup/backup_resiliency_test.go

```diff
@@ -1558,3 +1558,284 @@ var _ = Describe("{ScaleDownPxBackupPodWhileBackupAndRestoreIsInProgress}", func
+		})
+
+		Step("Registering source and destination clusters for backup", func() {
+			log.InfoD("Registering source and destination clusters for backup")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			err = CreateSourceAndDestClusters(orgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creating source cluster %s and destination cluster %s", SourceClusterName, destinationClusterName))
+			srcClusterStatus, err = Inst().Backup.GetClusterStatus(orgID, SourceClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", SourceClusterName))
+			dash.VerifyFatal(srcClusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", SourceClusterName))
+			srcClusterUid, err = Inst().Backup.GetClusterUID(ctx, orgID, SourceClusterName)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+			destClusterStatus, err = Inst().Backup.GetClusterStatus(orgID, destinationClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", destinationClusterName))
+			dash.VerifyFatal(destClusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", destinationClusterName))
+		})
+
+		Step("Taking backup of applications", func() {
+			log.InfoD("Taking backup of applications")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range appNamespaces {
+				backupNames := make([]string, 0)
+				log.Infof("Taking manual backup of application without rule's")
+				backupName = fmt.Sprintf("%s-%s-%v", BackupNamePrefix, namespace, time.Now().Unix())
+				err = CreateBackup(backupName, SourceClusterName, bkpLocationName, backupLocationUID,
+					[]string{namespace}, labelSelectors, orgID, srcClusterUid, "", "", "", "", ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Taking backup %s of application- %s", backupName, namespace))
+				backupNames = append(backupNames, backupName)
+
+				log.Infof("Taking schedule backup of application without rule's")
+				scheduleName = fmt.Sprintf("%s-%s-%v", BackupNamePrefix, namespace, time.Now().Unix())
+				schPolicyUid, _ = Inst().Backup.GetSchedulePolicyUid(orgID, ctx, periodicPolicyName)
+				err = CreateScheduleBackup(scheduleName, SourceClusterName, bkpLocationName, backupLocationUID, []string{namespace},
+					labelSelectors, orgID, "", "", "", "", periodicPolicyName, schPolicyUid, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verification of creating schedule backup with schedule name - %s", scheduleName))
+				schBackupName, err = GetFirstScheduleBackupName(ctx, scheduleName, orgID)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching the name of the first schedule backup - %s", schBackupName))
+				backupNames = append(backupNames, schBackupName)
+				scheduleNames = append(scheduleNames, scheduleName)
```

Comment:
I have a doubt about the behavior of this test:
You are creating both manual and scheduled backups, for both apps with and without rules.

Though, from the logic, it appears to me that, if we consider that you are backing up something with rules (like postgres), then it will be backed up 4 times, where two of the times are without using the rules despite the fact that postgres uses rules.
If we consider that you are backing up something without rules, then won't your logic do `Inst().Backup.GetRuleUid` for an app that doesn't have rules? In which case, it appears that you're ignoring the error value returned by `Inst().Backup.GetRuleUid`. **Though**, _this should still work_ as the `preRuleId` string will just be empty, which is the same as backing up without rules - and this is fine.

I guess what I'm saying is, if I'm understanding correctly, you can remove the logic where you create backup for app "without rules", since if an app doesn't have rules, the `preRuleId` and `postRuleId` will turn up empty, so it's enough to have 2 types of backups (mnual and schedule), where you look for pre and post rules in both cases and they will be used if the string is not `""`.

Reply:
The idea here was having seperate set of backup's. but currentky reverted the code to use normal backups due to some irregularities seen in updating rule from AppParameters.  We have a section called AppParameters where we define rules for any incoming apps so that there wont be any empty rules while creation of backup. 
The issue seen can be taken care when we test with mutiple apps.